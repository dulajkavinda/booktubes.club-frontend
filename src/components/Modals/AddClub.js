import * as React from 'react';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Modal from '@mui/material/Modal';
import TextField from '@mui/material/TextField';
import MenuItem from '@mui/material/MenuItem';

import { createClub } from '../../APIs/api.actions';

const style = {
	position: 'absolute',
	top: '50%',
	left: '50%',
	transform: 'translate(-50%, -50%)',
	width: 400,
	bgcolor: 'background.paper',

	boxShadow: 24,
	p: 4,
};

const currencies = [
	{
		value: 'Horror',
		label: 'Horror',
	},
	{
		value: 'Thriller',
		label: 'Thriller',
	},
	{
		value: 'Sci-Fi',
		label: 'Sci-Fi',
	},
	{
		value: 'Romance',
		label: 'Romance',
	},
];

const types = [
	{
		value: 'Free',
		label: 'Free',
	},
	{
		value: 'Paid',
		label: 'Paid',
	},
];

export default function AddClub({ open, setOpen }) {
	const handleClose = () => {
		console.log(currency, type, img, name, img);
		setOpen(false);
	};

	const [currency, setCurrency] = React.useState('Thriller');
	const [type, setType] = React.useState('Free');
	const [name, setName] = React.useState('');
	const [desc, setDesc] = React.useState('');
	const [img, setImg] = React.useState('');

	const handleChange = (event) => {
		setCurrency(event.target.value);
	};

	const handleTypeChange = (event) => {
		setType(event.target.value);
	};

	const handleSubmit = () => {
		createClub({
			name: name,
			desc: desc,
			img: img,
			type: type,
			currency: currency,
		}).then((response) => {
			if (response.code === 200) {
				setOpen(false);
			}
		});
	};
  
	return (
		<div>
			<Modal open={open} onClose={handleClose}>
				<Box sx={style}>
					<TextField
						fullWidth
						label='Club Name'
						id='fullWidth'
						onChange={(e) => setName(e.target.value)}
					/>
					<Box height={10} />
					<TextField
						fullWidth
						label='Club Description'
						id='fullWidth'
						onChange={(e) => setDesc(e.target.value)}
					/>
					<Box height={10} />
					<TextField
						fullWidth
						label='Club Image URL'
						id='fullWidth'
						onChange={(e) => setImg(e.target.value)}
					/>
					<Box height={15} />
					<TextField
						id='outlined-select-currency'
						select
						label='Club Category'
						value={currency}
						onChange={handleChange}
						helperText='Please select your club type'>
						{currencies.map((option) => (
							<MenuItem key={option.value} value={option.value}>
								{option.label}
							</MenuItem>
						))}
					</TextField>
					<Box height={15} />
					<TextField
						id='outlined-select-currency'
						select
						label='Club Type'
						value={type}
						onChange={handleTypeChange}
						helperText='Please select your club category'>
						{types.map((option) => (
							<MenuItem key={option.value} value={option.value}>
								{option.label}
							</MenuItem>
						))}
					</TextField>
					<Box height={15} />
					<Button onClick={handleSubmit} fullWidth variant='contained'>
						Create a Book Club
					</Button>
				</Box>
			</Modal>
		</div>
	);
}
