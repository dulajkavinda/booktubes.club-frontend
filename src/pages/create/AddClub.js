import React from "react";
import { Input } from "@chakra-ui/react";
import { Select } from "@chakra-ui/react";
import styles from "./AddClub.module.css";
import { Button, ButtonGroup } from "@chakra-ui/react";
import { useRouter } from "next/router";
import { createClub } from "../../APIs/api.actions";

import { useAuthState } from "react-firebase-hooks/auth";
import {auth} from "../../../firebase";

export default function AddClub() {
  const [category, setCategory] = React.useState("Thriller");
  const [type, setType] = React.useState("Free");
  const [name, setName] = React.useState("");
  const [desc, setDesc] = React.useState("");
  const [img, setImg] = React.useState("");

  const [loggedUser, loading,error] = useAuthState(auth);
  
  const router = useRouter();

  const handleChange = (event) => {
    setCategory(event.target.value);
  };

  const handleTypeChange = (event) => {
    setType(event.target.value);
  };

  const handleSubmit = () => {
    createClub({
      name: name,
      desc: desc,
      img: img,
      type: type,
      currency: category,
      admin : loggedUser
    }).then((response) => {
      if (response.code === 200) {
        router.push("/Dashboard/Dashboard");
      }
    });
  };

  console.log(loggedUser);
  return (
    <div className={styles.main}>
      <span className={styles.title}>Create your Own Club!</span>
      <div className={styles.form}>
        <Input
          onChange={(e) => setName(e.target.value)}
          mt={3}
          placeholder="Club Name"
        />
        <Input
          onChange={(e) => setDesc(e.target.value)}
          mt={3}
          placeholder="Club Description"
        />
        <Input
          onChange={(e) => setImg(e.target.value)}
          mt={3}
          placeholder="Club Image"
        />
        <Select
          onChange={handleTypeChange}
          value={type}
          mt={3}
          placeholder="Select a Club Type"
        >
          <option value="option1">Free</option>
          <option value="option2">Paid</option>
        </Select>
        <Select
          onChange={handleChange}
          value={category}
          mt={3}
          placeholder="Select a Category"
        >
          <option value="option1">Horror</option>
          <option value="option2">Thriller</option>
          <option value="option3">Sci-Fi</option>
          <option value="option3">Romance</option>
        </Select>
        <Button onClick={handleSubmit} mt={2} colorScheme="blue">
          Create a Club
        </Button>
      </div>
    </div>
  );
}
