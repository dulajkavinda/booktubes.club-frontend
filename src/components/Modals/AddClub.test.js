import React from "react";
import { shallow, mount } from "enzyme";

import AddClub from "./AddClub";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<AddClub />);
});

describe("add club basic tests", () => {
  it("is snapshot changed", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
