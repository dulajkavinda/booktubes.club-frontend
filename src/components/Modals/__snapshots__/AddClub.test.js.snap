// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add club basic tests is snapshot changed 1`] = `
<div>
  <ForwardRef(Modal)
    onClose={[Function]}
  >
    <ForwardRef(Box)
      sx={
        Object {
          "bgcolor": "background.paper",
          "boxShadow": 24,
          "left": "50%",
          "p": 4,
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": 400,
        }
      }
    >
      <ForwardRef(TextField)
        fullWidth={true}
        id="fullWidth"
        label="Club Name"
        onChange={[Function]}
      />
      <ForwardRef(Box)
        height={10}
      />
      <ForwardRef(TextField)
        fullWidth={true}
        id="fullWidth"
        label="Club Description"
        onChange={[Function]}
      />
      <ForwardRef(Box)
        height={10}
      />
      <ForwardRef(TextField)
        fullWidth={true}
        id="fullWidth"
        label="Club Image URL"
        onChange={[Function]}
      />
      <ForwardRef(Box)
        height={15}
      />
      <ForwardRef(TextField)
        helperText="Please select your club type"
        id="outlined-select-currency"
        label="Club Category"
        onChange={[Function]}
        select={true}
        value="Thriller"
      >
        <ForwardRef(MenuItem)
          key="Horror"
          value="Horror"
        >
          Horror
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Thriller"
          value="Thriller"
        >
          Thriller
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Sci-Fi"
          value="Sci-Fi"
        >
          Sci-Fi
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Romance"
          value="Romance"
        >
          Romance
        </ForwardRef(MenuItem)>
      </ForwardRef(TextField)>
      <ForwardRef(Box)
        height={15}
      />
      <ForwardRef(TextField)
        helperText="Please select your club category"
        id="outlined-select-currency"
        label="Club Type"
        onChange={[Function]}
        select={true}
        value="Free"
      >
        <ForwardRef(MenuItem)
          key="Free"
          value="Free"
        >
          Free
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)
          key="Paid"
          value="Paid"
        >
          Paid
        </ForwardRef(MenuItem)>
      </ForwardRef(TextField)>
      <ForwardRef(Box)
        height={15}
      />
      <ForwardRef(Button)
        fullWidth={true}
        onClick={[Function]}
        variant="contained"
      >
        Create a Book Club
      </ForwardRef(Button)>
    </ForwardRef(Box)>
  </ForwardRef(Modal)>
</div>
`;
